media-tv/xmltv/xmltv-0.6.1.ebuild

```diff
@@ -0,0 +1,287 @@
+	dev-perl/Lingua-Preferred
+	dev-perl/PerlIO-gzip
+	dev-perl/Term-ProgressBar
+	dev-perl/TermReadKey
+	dev-perl/XML-Parser
+	dev-perl/XML-TreePP
+	dev-perl/XML-Twig
+	dev-perl/XML-Writer
+	dev-perl/Unicode-String
+	virtual/perl-Memoize
+	virtual/perl-Storable
+	virtual/perl-IO-Compress
+"
+DEPEND="${RDEPEND}
+	ar? (
+		dev-perl/DateTime
+		dev-perl/HTML-Parser
+		dev-perl/HTML-Tree
+		dev-perl/HTTP-Cookies
+	)
+	ch-search? (
+		dev-perl/HTML-Tree
+		dev-perl/HTML-Parser
+		dev-perl/HTTP-Cookies
+		dev-perl/URI
+	)
+	dk? (
+		dev-perl/DateTime
+		dev-perl/IO-stringy
+	)
+	dtvla? (
+		dev-perl/DateTime
+		dev-perl/HTML-Parser
+		dev-perl/HTML-Tree
+		dev-perl/HTTP-Cookies
+	)
+	eu-dotmedia? (
+		dev-perl/HTTP-Cache-Transparent
+		dev-perl/IO-stringy
+	)
+	eu-epgdata? (
+		dev-perl/Archive-Zip
+		dev-perl/DateTime
+		dev-perl/DateTime-Format-Strptime
+		dev-perl/HTML-Parser
+		dev-perl/HTTP-Message
+	)
+	eu-xmltvse? (
+		dev-perl/HTTP-Cache-Transparent
+		dev-perl/IO-stringy
+	)
+	fi? (
+		dev-perl/HTML-Tree
+		dev-perl/LWP-Protocol-https
+		dev-perl/URI
+	)
+	fi-sv? (
+		dev-perl/DateTime
+		dev-perl/HTML-Tree
+		dev-perl/IO-stringy
+	)
+	fr? (
+		dev-perl/DateTime
+		dev-perl/DateTime-TimeZone
+		dev-perl/HTML-Parser
+		dev-perl/HTML-Tree
+	)
+	huro? (
+		dev-perl/HTML-Parser
+		dev-perl/HTML-Tree
+	)
+	il? ( dev-perl/DateTime )
+	is? (
+		dev-perl/HTML-Parser
+		dev-perl/HTML-Tree
+		dev-perl/URI
+		dev-perl/XML-DOM
+		dev-perl/XML-LibXSLT
+	)
+	it? (
+		dev-perl/HTML-Parser
+		dev-perl/HTML-Tree
+		dev-perl/URI
+	)
+	na-dd? ( dev-perl/SOAP-Lite )
+	na-dtv? (
+		dev-perl/DateTime
+		dev-perl/HTTP-Cookies
+		dev-perl/URI
+	 )
+	na-tvmedia? ( dev-perl/XML-LibXML )
+	pt-meo? (
+		dev-perl/DateTime
+		dev-perl/XML-LibXML
+	)
+	pt-vodafone? (
+		dev-perl/DateTime
+		dev-perl/URI
+		dev-perl/XML-LibXML
+	)
+	se-swedb? (
+		dev-perl/HTTP-Cache-Transparent
+		dev-perl/IO-stringy
+		dev-perl/XML-LibXML
+	)
+	se-tvzon? (
+		dev-perl/XML-LibXML
+		dev-perl/IO-stringy
+		dev-perl/HTTP-Cache-Transparent
+	)
+	tr? (
+		dev-perl/DateTime
+		dev-perl/HTTP-Cache-Transparent
+		dev-perl/HTTP-Cookies
+		dev-perl/URI
+	)
+	uk-bleb? (
+		dev-perl/Archive-Zip
+		dev-perl/IO-stringy
+	)
+	uk-tvguide? (
+		dev-perl/DateTime
+		dev-perl/HTML-Tree
+		dev-perl/HTTP-Cache-Transparent
+		dev-perl/HTTP-Cookies
+		dev-perl/URI
+	)
+	zz-sdjson? (
+		dev-lang/perl
+		dev-perl/DateTime
+		virtual/perl-Digest-SHA
+		dev-perl/HTTP-Message
+		dev-perl/LWP-Protocol-https
+		dev-perl/Try-Tiny
+	)
+	zz-sdjson-sqlite? (
+		dev-lang/perl
+		dev-perl/DateTime
+		dev-perl/DateTime-Format-ISO8601
+		dev-perl/DateTime-Format-SQLite
+		dev-perl/DateTime-TimeZone
+		dev-perl/DBD-SQLite
+		dev-perl/DBI
+		virtual/perl-Digest-SHA
+		dev-perl/File-HomeDir
+		dev-perl/File-Which
+		dev-perl/List-MoreUtils
+		dev-perl/LWP-UserAgent-Determined
+	)
+	tv-check? (
+		dev-perl/Tk
+		dev-perl/Tk-TableMatrix
+	)
+	tv-combiner? ( dev-perl/XML-LibXML )
+	tv-pick-cgi? ( dev-perl/CGI )
+"
```

Comment:
Yeah, you've definitely mixed up RDEPEND and DEPEND.

DEPEND is for "only required to build, you can remove them after that"

And then, on EAPI7, DEPEND is pretty much not used for perl stuff, BDEPEND is the right place for most perl stuff on EAPI7.

To the best of my knowledge, the only things that go in DEPEND on EAPI7 are things that need to be installed on the *target* ( as in, cross-compile target ), and thus, need to be built with the *target* architecture, in order for *this* package to build for the *target* architecture.

Reply:
Thank you for clarifying that.

Reply:
It still seems wrong though: 

- If the grabbers are installed
- And if the grabbers use those dependencies
- Then those dependencies *must* be in RDEPEND, or `--depclean` *will* break them.

Reply:
You are right. Move all to RDEPEND.